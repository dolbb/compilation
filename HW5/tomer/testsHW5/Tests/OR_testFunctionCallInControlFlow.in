void myPrint(int i){
	print("got ");
	printi(i);
	print("\n");
}
byte myPrint2(int i){
	print("got ");
	printi(i);
	print("\n");
	return 1b;
}
void main()
{
	int x = 1;
	byte b1;
	if(true)
		myPrint(x);
	if(true)
		byte b2 = myPrint2(x+1);
	if(true)
		b1 = myPrint2(x+2);

	x = x + 3;//4
	if(true)
		{myPrint(x);}
	if(true)
		{byte b3 = myPrint2(x+1);}
	if(true)
		{b1 = myPrint2(x+2);}

	x = x + 3;//7
	if(true)
		myPrint(x);
	else
		int t;
	if(true)
		byte b4 = myPrint2(x+1);
	else
		int t;
	if(true)
		b1 = myPrint2(x+2);
	else
		int t;

	x = x + 3;//10
	if(true)
		{myPrint(x);}
	else
		int t;
	if(true)
		{byte b5 = myPrint2(x+1);}
	else
		int t;
	if(true)
		{b1 = myPrint2(x+2);}
	else
		int t;

	x = x + 3;//13
	if(false)
		int t;
	else
		myPrint(x);
	if(false)
		int t;
	else
		byte b6 = myPrint2(x+1);
	if(false)
		int t;
	else
		b1 = myPrint2(x+2);

	x = x + 3;//16
	if(false)
		int t;
	else
		{myPrint(x);}
	if(false)
		int t;
	else
		{byte b7 = myPrint2(x+1);}
	if(false)
		int t;
	else
		{b1 = myPrint2(x+2);}
	b1 = 0b;
	x = x + 3;//19
	while(b1 == 0)
		b1 = myPrint2(x);
	b1 = 0b;
	while(b1 == 0)
		{b1 = myPrint2(x+1);}
	while(true)
	{
		myPrint(x+2);
		break;
	}
	while(b1 == 1b)
	{
		byte b69 = myPrint2(x+3);
		b1 = b1 - b69;
	}
}