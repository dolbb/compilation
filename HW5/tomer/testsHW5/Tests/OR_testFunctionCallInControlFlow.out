Loaded: ./exceptions.s
got 1
got 2
got 3
got 4
got 5
got 6
got 7
got 8
got 9
got 10
got 11
got 12
got 13
got 14
got 15
got 16
got 17
got 18
got 19
got 20
got 21
got 22
